### PR TITLE
Move Java classes under java top-level dir

### PR DESCRIPTION
**Pull request by tiziano88**
Create per-directory BUILD files (not sure why this was not done before).

Add scripts to build and run Android Docker image.

Update CI to use new scripts.

Update Android Docker image to a later version of Debian and JDK, as
well as newer Android SDK and API.

Ref #2533

### CODE REVIEW

File: xtask/src/main.rs [hunk 1]
```diff
@@ -745,33 +745,13 @@ fn run_cargo_clean() -> Step {
...
-            [
-                "build",
-                "--",
-                "//oak_functions/client/java/...:all",
-                "//oak_functions/examples/...:all",
-                "//remote_attestation/java/...:all",
-                "//java/...:all",
-            ],
-        ),
+        command: Cmd::new("bazel", ["build", "--", "//java/...:all"]),
```
Comment (rbehjati) [inline xtask/src/main.rs:748]: Nice!

**Review by rbehjati — APPROVED** [review]
Nice. Thanks.

File: android.Dockerfile [hunk 2]
```diff
@@ -15,7 +16,7 @@ RUN apt-get --yes update && \
     libmpfr-dev \
     libncurses5 \
     ocaml-nox \
-    openjdk-8-jdk \
+    openjdk-11-jdk \
```
Comment (ipetr0v) [inline android.Dockerfile:19]: wow
Android now supports Java 11 ?

File: java/src/main/java/com/google/oak/functions/weather_lookup_client/BUILD [hunk 3]
```diff
@@ -21,13 +21,11 @@ package(
...
-    srcs = [
-        "src/Main.java",
-    ],
-    main_class = "com.google.oak.functions.examples.weather_lookup.client.Main",
+    name = "weather_lookup_client",
+    srcs = ["Main.java"],
+    main_class = "com.google.oak.functions.weather_lookup_client.Main",
     deps = [
-        "//oak_functions/client/java:client",
+        "//java/src/main/java/com/google/oak/functions/client",
```
Comment (ipetr0v) [inline java/src/main/java/com/google/oak/functions/weather_lookup_client/BUILD:28]: Why is `java` appearing in the path 2 times?

**Review by ipetr0v — APPROVED** [review]

File: java/src/main/java/com/google/oak/functions/weather_lookup_client/BUILD (hunk 3 above)
Comment (tiziano88) [reply java/src/main/java/com/google/oak/functions/weather_lookup_client/BUILD:28]: That's the full path to the file at the moment

Comment (rbehjati) [reply java/src/main/java/com/google/oak/functions/weather_lookup_client/BUILD:28]: Yes. The top-level java is our naming to group all the java code in one place. The file structure inside the top-level java, including `src/main/java` and `src/test/java` is the conventional way of organizing a java project.

Comment (tiziano88) [reply java/src/main/java/com/google/oak/functions/weather_lookup_client/BUILD:28]: Thanks for the added context @rbehjati !

**Pull request merged**